packages/@aws-cdk/aws-applicationautoscaling/lib/schedule.ts

```diff
@@ -17,11 +17,11 @@ export abstract class Schedule {
    * Construct a schedule from an interval and a time unit
    */
   public static rate(duration: Duration): Schedule {
-    const validDurationUnit = ['minute', 'minutes', 'hour', 'hours', 'day', 'days'];
-    if (!validDurationUnit.includes(duration.unitLabel())) {
-      throw new Error("Allowed unit for scheduling is: 'minute', 'minutes', 'hour', 'hours', 'day' or 'days'");
-    }
     if (duration.isUnresolved()) {
+      const validDurationUnit = ['minute', 'minutes', 'hour', 'hours', 'day', 'days'];
+      if (!validDurationUnit.includes(duration.unitLabel())) {
+        throw new Error("Allowed units for scheduling are: 'minute', 'minutes', 'hour', 'hours', 'day' or 'days'");
+      }
       return new LiteralSchedule(`rate(${duration.formatTokenToNumber()})`);
     }
```

Comment:
```suggestion
    if (duration.isUnresolved()) {
          return new LiteralSchedule(`rate(${duration.formatTokenToNumber()})`)
    }
    
      const validDurationUnit = ['minute', 'minutes', 'hour', 'hours', 'day', 'days'];
      if (!validDurationUnit.includes(duration.unitLabel())) {
        throw new Error("Allowed units for scheduling are: 'minute', 'minutes', 'hour', 'hours', 'day' or 'days'");
      }
    
    
```
```suggestion
    if (duration.isUnresolved()) {
      const validDurationUnit = ['minute', 'minutes', 'hour', 'hours', 'day', 'days'];
      if (!validDurationUnit.includes(duration.unitLabel())) {
        throw new Error("Allowed units for scheduling are: 'minute', 'minutes', 'hour', 'hours', 'day' or 'days'");
      }
      return new LiteralSchedule(`rate(${duration.formatTokenToNumber()})`);
    }
```

Reply:
I think there is still a problem here. If duration is unresolved, it will go through the validation.